src/cluster/placement/types.go

```diff
@@ -440,14 +440,14 @@ const (
 // Storage provides read and write access to placement.
 type Storage interface {
 	// Set writes a placement.
-	Set(p Placement) error
+	Set(p Placement) (int, error)
 
 	// CheckAndSet writes a placement if the current version
 	// matches the expected version.
-	CheckAndSet(p Placement, version int) error
+	CheckAndSet(p Placement, version int) (int, error)
 
 	// SetIfNotExist writes a placement.
-	SetIfNotExist(p Placement) error
+	SetIfNotExist(p Placement) (int, error)
```

Comment:
Since you are here, could you change
`Placement() (Placement, int, error)` to `Placement() (Placement,error)`?